test/FunctionalTests/Microsoft.OData.Edm.Tests/Csdl/CsdlWriterTests.TargetPath.cs

```diff
@@ -0,0 +1,418 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="CsdlWriterTests.TargetPath.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.OData.Edm.Csdl;
+using Microsoft.OData.Edm.Vocabularies;
+using Xunit;
+
+namespace Microsoft.OData.Edm.Tests.Csdl
+{
+    public partial class CsdlWriterTests
+    {
+        [Fact]
+        public void ShouldWriteAnnotationForEntitySetProperty()
+        {
+            // Arrange
+            EdmModel model = new EdmModel();
+            EdmEntityType customer = new EdmEntityType("NS", "Customer");
+            customer.AddKeys(customer.AddStructuralProperty("Id", EdmCoreModel.Instance.GetInt32(false)));
+            customer.AddStructuralProperty("Name", EdmPrimitiveTypeKind.String, isNullable: false);
+            model.AddElement(customer);
+
+            EdmEntityContainer container = new EdmEntityContainer("NS", "Default");
+            EdmEntitySet entitySet = new EdmEntitySet(container, "Customers", customer);
+            container.AddElement(entitySet);
+            model.AddElement(container);
+
+            IEdmProperty nameProperty = customer.DeclaredProperties.Where(x => x.Name == "Name").FirstOrDefault();
+
+            EdmTargetPath targetPath = new EdmTargetPath(container, entitySet, nameProperty);
+
+            EdmTerm term = new EdmTerm("NS", "MyTerm", EdmCoreModel.Instance.GetString(true));
+            model.AddElement(term);
+            EdmVocabularyAnnotation annotation = new EdmVocabularyAnnotation(targetPath, term, new EdmStringConstant("Name OutOfLine MyTerm Value"));
+            annotation.SetSerializationLocation(model, EdmVocabularyAnnotationSerializationLocation.OutOfLine);

```

Comment:
>EdmVocabularyAnnotationSerializationLocation.OutOfLine); [](http://example.com/codeflow?start=55&length=56)

what about if customer uses "EdmVocabularyAnnotationSerializationLocation.InLine"?

Reply:
@xuzhg 
The current behaviour is that the annotation is not written.
I believe that an `inline annotation` is written within a `model element` so we should be targeting a `model element`. If we target an `IEdmTargetPath` (path to a model element), we should not be able to write the annotation.
cc @mikepizzo @gathogojr

Reply:
> @xuzhg The current behaviour is that the annotation is not written. I believe that an `inline annotation` is written within a `model element` so we should be targeting a `model element`. If we target an `IEdmTargetPath` (path to a model element), we should not be able to write the annotation. cc @mikepizzo @gathogojr

We should throw exception if cusomter call 
`annotation.SetSerializationLocation(model, EdmVocabularyAnnotationSerializationLocation.InLine);`

Reply:
Fixed here https://github.com/OData/odata.net/pull/2769/commits/c8c1da331ec535b23441baaed76498ecbec47a57